lib/strategies/webapp-strategy.js

```diff
@@ -344,12 +345,20 @@ function getTokens(req, options, strategy) {
 		code: req.query.code
 	};
 
+	function cb(err, user, info) {
+		if (err) {
+			return strategy.fail;
```

Comment:
I know that in the existing code we call fail like this but I think it makes more sense here to call it with the error like strategy.fail(err)